### DISCUSSION TITLE
simple_cycles taking an extremely long time?

### DISCUSSION THREAD

**Discussion by danben** [networkx/networkx#5170] [Q&A] [answered]
I have a MultiDiGraph of 469 nodes and 1112 edges. I can run simple_cycles on this and finish in ~10s.

Adding another 30 edges seems to increase the running time dramatically - I let it run for 12 hours before giving up. Even assuming these 30 edges were placed in the worst possible spots (which tbh they likely are), this feels like a much higher increase than I would have expected (though I'm far from a graph theory expert so I could just be wrong here).

Are there any kinds of mistakes I could have made that would cause the running time of simple_cycles to blow up dramatically or even fail to terminate? Or is it possible that somehow these extra edges actually did increase the running time by this much?

**Comment by danben** [comment, marked as answer]
I think I've convinced myself that I just added a lot more cycles. I couldn't figure out how to delete my question, sorry.

**Comment by rossbar** [comment]
No worries @danben - I'm happy to delete this for you, but I've never deleted a discussion before so I'm not quite sure what happens. I'm not sure whether discussions remain "visible" (like a closed issue) once they've been deleted... do you want me to try anyways?

**Reply by danben** [reply]
Thanks @rossbar - doesn't matter much to me, I just didn't want to pollute the forum

**Reply by rossbar** [reply]
In that case I will just mark it as answered - you're definitely not the only user who has run into this, so it may be useful to others :+1:
